models/classes/LtiProvider/Validation/LtiProviderValidationService.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoLti\models\classes\LtiProvider\Validation;
+
+use InvalidArgumentException;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoLti\models\classes\LtiProvider\LtiProviderFieldsMapper;
+
+class LtiProviderValidationService extends ConfigurableService
+{
+    /**
+     * @var array
+     */
+    private $errors = [];
+
+    /**
+     * @throw InvalidArgumentException
+     */
+    public function validateArray(string $schema, array $data): void
+    {
+        $this->errors = [];
```

Comment:
Instead of aggregating errors we could throw exceptions maybe? WDYT?

Reply:
Then it would stop on the first error, but I'd prefer to see all of them at once